src/Trakx.CoinGecko.ApiClient/Configuration/ApiClientExtensions.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Polly.Contrib.WaitAndRetry;
+using Polly.Extensions.Http;
+using Trakx.Common.ApiClient;
+using Trakx.Common.DateAndTime;
+using Trakx.Common.Logging;
+
+namespace Trakx.CoinGecko.ApiClient;
+
+public static partial class ApiClientExtensions
+{
+    public static IServiceCollection AddCoinGeckoClient(
+        this IServiceCollection services, IConfiguration configuration)
+    {
+        var section = configuration.GetSection(nameof(CoinGeckoApiConfiguration));
+        services.Configure<CoinGeckoApiConfiguration>(section);
+
+        var typedConfig = section.Get<CoinGeckoApiConfiguration>()!;
```

Comment:
Could you have used your 
Trakx.Common.Configuration.ConfigurationExtensions.GetConfiguration<TConfiguration>(this IConfiguration configuration) ?

Reply:
Ah, yes, I'll add it in the next PR.